api/v1beta1/inference_config_validation.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kaito-project/kaito/pkg/k8sclient"
+	"github.com/kaito-project/kaito/pkg/utils"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// InferenceConfig represents the structure of the inference configuration
+type InferenceConfig struct {
+	VLLM map[string]string `yaml:"vllm"`
+	// Other fields can be added as needed
+}
+
+func (i *InferenceSpec) validateConfigMap(ctx context.Context, namespace, cm_name, instanceType string) (errs *apis.FieldError) {
```

Comment:
In golang, we don't use this naming style. Just use `cmName` instead of `cm_name`.